### PR TITLE
feat: workflows editor code split

### PR DESCRIPTION
**Pull request by Druthi**
## Description
Code split for workflows code editor setup

#### PR fixes following issue(s)
Fixes [#2873](https://github.com/appsmithorg/appsmith-ee/issues/2873)

#### Type of change

- New feature (non-breaking change which adds functionality)

## Testing

- [ ] Manual
- [ ] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Comment by ayushpahwa** [comment]
Adding skip testplan label since this is code split for EE feature behind flag

File: app/client/src/pages/Editor/Explorer/JSActions/JSActionEntity.tsx [hunk 1]
```diff
@@ -62,15 +65,16 @@ export const ExplorerJSCollectionEntity = memo(
 
     const isFeatureEnabled = useFeatureFlag(FEATURE_FLAG.license_gac_enabled);
 
-    const canDeleteJSAction = getHasDeleteActionPermission(
-      isFeatureEnabled,
-      jsActionPermissions,
+    const isJsActionForWorkflowCreation = useSelector((state) =>
+      getIsJsActionForWorkflowCreation(state, jsAction.id),
```
Comment (ankitakinger) [inline app/client/src/pages/Editor/Explorer/JSActions/JSActionEntity.tsx:69]: I don't think this is thr right approach to do this. Ideally you should have removed the permissions outside this component and sent as props using the correct permission checks. Are we never planning to add GAC for workflows?

**Comment by ayushpahwa** [comment]
Breaking down this PR to smaller chunks

File: app/client/src/ce/selectors/applicationSelectors.tsx [hunk 2]
```diff
@@ -39,18 +41,21 @@ const fuzzySearchOptions = {
  *    packages: []
  *  }
  */
-const injectPackagesToWorkspacesList = (
+const injectPackagesAndWOrkflowsToWorkspacesList = (
```
Comment (ashit-rath) [inline app/client/src/ce/selectors/applicationSelectors.tsx:44]: `injectPackagesAndWOrkflowsToWorkspacesList` -> `injectPackagesAndWorkflowsToWorkspacesList`

File: app/client/src/pages/Editor/JSEditor/Form.tsx [hunk 3]
```diff
@@ -164,20 +165,30 @@ function JSEditorForm({ jsCollection: currentJSCollection }: Props) {
     }
   }, [hash]);
 
+  const isJsActionForWorkflowCreation = useSelector((state) =>
```
Comment (ashit-rath) [inline app/client/src/pages/Editor/JSEditor/Form.tsx:168]: Let's not directly add workflow specific code in the JSEditor itself.

I remember discussing to hold off working on the JSEditor. Has something changed?
cc: @srix

File: app/client/src/sagas/EvaluationsSaga.ts [hunk 4]
```diff
@@ -605,7 +606,15 @@ function* evaluationChangeListenerSaga(): any {
   const initAction: EvaluationReduxAction<unknown> = yield take(
     FIRST_EVAL_REDUX_ACTIONS,
   );
-  yield call(waitForWidgetConfigBuild);
+
+  const isCurrentEditorWorkflowType = yield select(
+    getIsCurrentEditorWorkflowType,
+  );
+
+  if (!isCurrentEditorWorkflowType) {
```
Comment (ashit-rath) [inline app/client/src/sagas/EvaluationsSaga.ts:614]: @ApekshaBhosale Can you verify this?

File: app/client/src/utils/JSPaneUtils.tsx [hunk 5]
```diff
@@ -232,6 +234,25 @@ export const createDummyJSCollectionActions = (
       clientSideExecution: true,
     },
   ];
+
+  if (from === "WORKFLOW_CREATION") {
```
Comment (ashit-rath) [inline app/client/src/utils/JSPaneUtils.tsx:238]: Why are we not splitting this file to introduce this? This change should be present in the EE

File: app/client/src/pages/Editor/JSEditor/Form.tsx (hunk 3 above)
Comment (ayushpahwa) [reply app/client/src/pages/Editor/JSEditor/Form.tsx]: Regarding the JS changes, Druthi had already done these (she didn't have context of our conversation). I have removed them already from the codebase this morning and will add in a separate PR.

File: app/client/src/ce/selectors/applicationSelectors.tsx (hunk 2 above)
Comment (ayushpahwa) [reply app/client/src/ce/selectors/applicationSelectors.tsx:44]: Fixed

File: app/client/src/utils/JSPaneUtils.tsx (hunk 5 above)
Comment (ayushpahwa) [reply app/client/src/utils/JSPaneUtils.tsx:238]: Agreed, removing this change since we are planning to add it in jseditor code split

**Review by ashit-rath — APPROVED** [review]

File: app/client/src/ce/constants/WorkflowConstants.ts [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+export interface Workflow {
+  id: ID;
+  name: string; // Name of the workflow.
+  icon: string;
+  color: string;
+  workspaceId: ID; // ID of the workspace where the workflow is created.
+  modifiedBy: string;
+  modifiedAt: string;
+  userPermissions: string[];
+  new: boolean;
```
Comment (ankitakinger) [inline app/client/src/ce/constants/WorkflowConstants.ts:13]: are you sure you are able to use the key `new` here? Since its a pre-defined keyword.

Comment (ayushpahwa) [reply app/client/src/ce/constants/WorkflowConstants.ts:13]: Yeah, this is used in applications also. Ref: https://github.com/appsmithorg/appsmith/blob/04cbb8591521a1f7444523b7f4d5353aca5f181a/app/client/src/ce/api/ApplicationApi.tsx#L212

**Review by ashit-rath — APPROVED** [review]

**Review by ankitakinger — APPROVED** [review]
LGTM

**Review by ashit-rath — APPROVED** [review]

**Pull request merged**